### PR TITLE
[Doc] add doc for ngram index

### PR DESCRIPTION
**Pull request by before-Sunrise**
## What I'm doing:

Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [x] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

**Comment by DanRoscigno** [comment]
@mergify backport branch-3.3

File: docs/en/sql-reference/sql-functions/string-functions/ngram_search.md [hunk 1]
```diff
@@ -6,40 +6,46 @@ displayed_sidebar: "English"
...
+Returns a value describing how similar these two strings are. The range of returned value is between 0 and 1. The larger this value, the more similar the two strings are.
 
 ## Examples
 
-```Plain Text
--- haystack and needle are const value
-mysql> select ngram_search("chinese","china");
+```SQL
+-- The values of haystack and needle are const values.
+mysql> select ngram_search("chinese","china",4);
```
Comment (wangsimo0) [inline docs/en/sql-reference/sql-functions/string-functions/ngram_search.md:40]: 英文文档尽量别用china和chinese

File: docs/zh/sql-reference/sql-functions/string-functions/ngram_search.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+- `haystack`: 必填项，要比较的第一个字符串。必须是 VARCHAR 值，可以是列名或常量值。
+- `needle`: 必填项，要比较的第二个字符串。必须是 VARCHAR 值，只能是常量值。
+
+    :::tip
+  
+    - `needle` 值的长度不能大于 2^15，否则会抛出错误。
+    - 如果 `haystack` 值的长度大于 2^15，则此函数将返回 0。
+    - 如果 `haystack` 或 `needle` 值的长度小于 `gram_num`，则此函数将返回 0。
+
```
Comment (wangsimo0) [inline docs/zh/sql-reference/sql-functions/string-functions/ngram_search.md:27]: - 如果 `haystack` 列上存在 ngram 索引，可以加速此函数的运算速度。

File: docs/zh/table_design/indexes/Ngram_Bloom_Filter_Index.md [hunk 3]
```diff
@@ -0,0 +1,126 @@
+---
+displayed_sidebar: "Chinese"
+---
+
+# N-Gram bloom filter 索引
+
+N-Gram bloom filter 索引是一种特殊的 [Bloom filter 索引](./Bloomfilter_index.md)，只适用于字符串（`STRING`、`CHAR` 或 `VARCHAR`）类型的列。N-Gram bloom filter 索引与 Bloom filter 索引的区别在于，N-Gram bloom filter 索引会先根据对字符串进行分词，然后将结果子串写入 Bloom filter 索引。
```
Comment (wangsimo0) [inline docs/zh/table_design/indexes/Ngram_Bloom_Filter_Index.md:7]: 最好加一下适用场景，例如：常被用于加速 `like` 查询或 `ngram_search`、 `ngram_search_case_insensitive` 函数。

File: docs/zh/table_design/indexes/Ngram_Bloom_Filter_Index.md [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+
+```SQL
+"Tech", "echn", "chni", "hnic", "nica", "ical"
+```
+
+然后将这六个子串中的每一个子串写入 Bloom filter 索引。
+
+## 使用说明
+
+- 对于明细表或主键表，您可以为所有列（字符串类型）创建 N-Gram bloom filter 索引。对于聚合表或更新，您只能为 Key 列（字符串类型）创建 N-Gram bloom filter 索引。
```
Comment (wangsimo0) [inline docs/zh/table_design/indexes/Ngram_Bloom_Filter_Index.md:19]: 对于聚合表或更新--少个表字

File: docs/en/sql-reference/sql-functions/string-functions/ngram_search.md [hunk 5]
```diff
@@ -6,40 +6,51 @@ displayed_sidebar: "English"
...
+- `needle`: required, the second string to compare. It must be a VARCHAR value. It can only be a const value.
+
+  :::tip
 
-> - needle's size can not be larger than 2^15, otherwise error will be thrown.
-> - if haystack's size is larger than 2^15, this function will return 0.
-> - If haystack or needle's size is smaller than N(which is the size of gram, right now is 4), then this function will return 0
+  - The length of the `needle` value can not be larger than 2^15. Otherwise an error will be thrown.
+  - If the length of the `haystack` value is larger than 2^15, this function will return 0.
+  - If the length of the `haystack` or `needle` value is smaller than `gram_num`, this fucntion will return 0.
```
Comment (EsoragotoSpirit) [inline docs/en/sql-reference/sql-functions/string-functions/ngram_search.md:31]: fucntion --> function

**Review by evelynzhaojie — DISMISSED** [review]

File: docs/en/table_design/indexes/Ngram_Bloom_Filter_Index.md [hunk 6]
```diff
@@ -0,0 +1,128 @@
+---
+displayed_sidebar: "English"
+
+---
+
+# N-gram bloom filter index
+
+The N-gram bloom filter index is a special [Bloom filter index](./Bloomfilter_index.md) which is typically used to accelerate the LIKE queries and the caculation speed of the `ngram_search` and `ngram_search_case_insensitive` functions.
```
Comment (EsoragotoSpirit) [inline docs/en/table_design/indexes/Ngram_Bloom_Filter_Index.md:8]: caculation --> calculation

File: docs/en/table_design/indexes/Ngram_Bloom_Filter_Index.md [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+
+The N-gram bloom filter index is only suitable for string (`STRING`, `CHAR`, or `VARCHAR`) type columns. The difference between the N-gram bloom filter index and Bloom filter index is that the N-gram bloom filter index first tokenizes the strings and then writes the resulting substrings into the Bloom filter index. For example, a certain indexed column value is a string `Technical`. For a traditional Bloom filter index, the entire string `Technical` would be directly written into the index. However, for an N-gram bloom filter index with a specified `gram_num` of `4`，the string `Technical` would be first tokenized into the following substrings:
```
Comment (EsoragotoSpirit) [inline docs/en/table_design/indexes/Ngram_Bloom_Filter_Index.md:10]: `gram_num` of `4`，中文逗号

File: docs/en/table_design/indexes/Ngram_Bloom_Filter_Index.md [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+
+- Execute the following statement to add a new N-gram bloom filter index `new_index_name` for the column `k1` to the table `table1`.
+
+  ```SQL
+  ALTER TABLE table1 
+  ADD INDEX new_index_name(k1) USING NGRAMBF ("gram_num" = "4", 
+                                              "bloom_filter_fpp" = "0.05") COMMENT '';
+  ```
+
+- The following statement to remove the N-gram bloom filter index `new_index_name` from the table `table1`.
```
Comment (EsoragotoSpirit) [inline docs/en/table_design/indexes/Ngram_Bloom_Filter_Index.md:78]: Execute the following statement to remove 

or

The following statement is to remove

File: docs/en/table_design/indexes/Ngram_Bloom_Filter_Index.md [hunk 9]
```diff
@@ -0,0 +1,128 @@
...
+    INDEX index_name (k2) USING NGRAMBF ("gram_num" = "4",
+                                         "bloom_filter_fpp" = "0.05",
+                                         "case_sensitive" = "false") COMMENT ''
+)
+ENGINE = olap
+AGGREGATE KEY(k1, k2)
+DISTRIBUTED BY HASH(k1);
+```
+
+If the index is already created and its paramter `case_sensitive` is set to be `true`, you can alter the paramter of this index to `false` by using:
```
Comment (EsoragotoSpirit) [inline docs/en/table_design/indexes/Ngram_Bloom_Filter_Index.md:121]: paramter --> parameter

有两个

File: docs/en/table_design/indexes/Ngram_Bloom_Filter_Index.md [hunk 10]
```diff
@@ -0,0 +1,128 @@
...
+    INDEX index_name (k2) USING NGRAMBF ("gram_num" = "4",
+                                         "bloom_filter_fpp" = "0.05") COMMENT ''
+)
+ENGINE = olap
+AGGREGATE KEY(k1, k2)
+DISTRIBUTED BY HASH(k1);
+
+```
+
+N-gram bloom filter index related parameters:
```
Comment (EsoragotoSpirit) [inline docs/en/table_design/indexes/Ngram_Bloom_Filter_Index.md:44]: index-related

**Review by EsoragotoSpirit — DISMISSED** [review]

**Review by EsoragotoSpirit — APPROVED** [review]

**Review by hellolilyliuyi — APPROVED** [review]

**Pull request merged**